main.go

```diff
@@ -197,3 +206,16 @@ func setLoggingLevel(debug bool, logLevel string) {
 		log.SetLevel(level)
 	}
 }
+
+func startWebhookServer(webhookHandler *utils.WebhookHandler) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/api/webhook", webhookHandler.Handler)
+	go func() {
+		setupLog.Info("Starting webhook server")
+		err := http.ListenAndServe(":7000", mux)
+		if err != nil {
+			setupLog.Error(err, "failed to start webhook server")
+			os.Exit(1)
+		}
+	}()
+}
```

Comment:
Hmm, is it a good idea to hardcode the port here as other port bindings are configurable: https://github.com/argoproj-labs/applicationset/blob/master/main.go#L84-L85
```golang
	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
	flag.StringVar(&probeBindAddr, "probe-addr", ":8081", "The address the probe endpoint binds to.")
```
Also it would be nice to make the webhook handler optional. This introduces a hard dependency on the ConfigMap `argocd-cm` which makes it really hard to find a good solution for the helm chart update to 0.3.0:
https://github.com/argoproj/argo-helm/pull/1070

Reply:
Hi @mkilchhofer, thanks for bringing this up. I agree the webhook address should be configurable. I've raised a PR for this https://github.com/argoproj-labs/applicationset/pull/450. It also makes the webhook handler optional by removing the hard exit.

Reply:
Cool. Also worked on this :-D but you were faster, thanks. 🙏